flake.nix

```diff
@@ -141,7 +141,8 @@
         megaparsec = sdistTarball haskellPackages.megaparsec;
         megaparsec-tests = sdistTarball haskellPackages.megaparsec-tests;
       };
-
+      haskellLanguageServer = pkgs.haskell.lib.overrideCabal haskellPackages.haskell-language-server
```

Comment:
Why is this necessary?

Reply:
Nixshell provides HLS, so `code .` in shell gets proper HLS for GHC.

Reply:
My question was rather specifically why `enableSharedExecutables = true` should be set, why not use `haskellLanguageServer` as it is?